dev-scripts/test-migration-e2e.py

```
@@ -0,0 +1,640 @@
+                pass
+            p.wait(timeout=5)
+    f.close()
+    wait_port_close(PORT, timeout=15)
+
+
+def wait_ready(timeout=120):
+    deadline = time.time() + timeout
+    url = f"http://127.0.0.1:{PORT}/nix-cache-info"
+    while time.time() < deadline:
+        try:
+            with urllib.request.urlopen(url, timeout=2) as r:
```

Ruff (0.15.13)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)